src/read/cfi.rs

```diff
@@ -1016,12 +1016,22 @@ where
     R: Reader,
     Section: UnwindSection<R>,
 {
-    let offset = input.offset_from(section.section());
-    let (length, format) = input.read_initial_length()?;
+    let (offset, length, format) = loop {
+        let offset = input.offset_from(section.section());
+        let (length, format) = input.read_initial_length()?;
 
-    if Section::length_value_is_end_of_entries(length) {
-        return Ok(None);
-    }
+        if Section::length_value_is_end_of_entries(length) {
+            return Ok(None);
+        }
+
+        if length.into_u64() != 0 {
+            break (offset, length, format);
+        }
+
+        // Same behaviour as in readelf
+        // https://github.com/bminor/binutils-gdb/blob/master/binutils/dwarf.c#L7820
+        input.skip_null()?;
```

Comment:
It seems this can only work correctly if the real length doesn't start with a zero?

Reply:
If the length start with a zero, then `read_initial_length()` will return a format Dwarf32.
So adding ` || format != Format::Dwarf32` in the condition to break shoud be ok.

Reply:
I mean if the real length is [0x00, 0x10, 0x00, 0x00], then this will skip the first 0x00 and try to read the length starting at the 0x10. Or maybe I am misunderstanding what this is doing. I'll have a look at the sample file tomorrow.

Reply:
I'm pretty sure my understanding is correct. I think the simplest fix is to delete this line so we loop around to try reading a length again, since CFI entries must be aligned to the address size. This also avoids the need to add the `skip_null` method.